shell/platform/linux/public/flutter_linux/fl_plugin_registrar.h

```diff
@@ -0,0 +1,53 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_LINUX_FL_PLUGIN_REGISTRAR_H_
+#define FLUTTER_SHELL_PLATFORM_LINUX_FL_PLUGIN_REGISTRAR_H_
+
+#if !defined(__FLUTTER_LINUX_INSIDE__) && !defined(FLUTTER_LINUX_COMPILATION)
+#error "Only <flutter_linux/flutter_linux.h> can be included directly."
+#endif
+
+#include <glib-object.h>
+#include "fl_binary_messenger.h"
+#include "fl_plugin.h"
+
+G_BEGIN_DECLS
+
+G_DECLARE_FINAL_TYPE(FlPluginRegistrar,
+                     fl_plugin,
+                     FL,
+                     PLUGIN_REGISTRAR,
+                     GObject)
+
+/**
+ * FlPluginRegistrar:
+ *
+ * #FlPluginRegistrar is FIXME
+ */
+
+/**
+ * fl_plugin_registrar_get_messenger:
+ * @registrar: an #FlPluginRegistrar
+ *
+ * Get the messenger this plugin can communicate with.
+ *
+ * Returns: an #FlBinaryMessenger
+ */
+FlBinaryMessenger* fl_plugin_registrar_get_messenger(
+    FlPluginRegistrar* registrar);
+
```

Comment:
Do you want to go ahead and add the view now? We'll need it for several plugins.

Reply:
Yes, I was still working out what GTK widgets we need. We'll need a container (vertical box) to put the menubar in, not sure if we need anything else.